eth_account/_utils/signing.py

```diff
@@ -40,14 +41,25 @@ def sign_transaction_dict(eth_key, transaction_dict):
 
 
 # watch here for updates to signature format: https://github.com/ethereum/EIPs/issues/191
-def signature_wrapper(message, version=b'E'):
-    assert isinstance(message, bytes)
-    if version == b'E':
+@curry
+def signature_wrapper(message, signature_version, version_specific_data):
+    if not isinstance(message, bytes):
+        raise TypeError("Message is not of the type {}, expected bytes".format(type(message)))
+
+    if signature_version == b'E':
```

Comment:
The `b'E'` should probably be encoded as a constant with a semantically meaningful name rather than using the literal inline here.  Same with the null byte used below for the other version.